src/passes/Memory64Lowering.cpp

```diff
@@ -79,9 +79,31 @@ struct Memory64Lowering : public WalkerPass<PostWalker<Memory64Lowering>> {
     if (memory->is64()) {
       wrapAddress64(curr->delta, curr->memory);
       auto* size = static_cast<Expression*>(curr);
-      extendAddress64(size, curr->memory);
+      // MemoryGrow returns -1 in the case of failure
+      // so we need to handle that
+      // local.set $tmp memory.grow X
+      // local.get
+      // i32.const -1
+      // if i32.eq
+      //   i64.const -1
+      // else
+      //   local.get
+      //   i32.extend
```

Comment:
Can you structure this example code the same way Binaryen would print it? You can also make this slightly shorter by using a `local.tee` at the beginning if you want.

Reply:
Is that true for the example code or also for the `builder` code?

Reply:
I actually don't know how binaryen would print this.. can you help me out?

Reply:
Actually I think I got it..

Reply:
> Can you structure this example code the same way Binaryen would print it?

Heh I was just writing that out in a parallel comment, see below.

Reply:
Using a `local.tee` makes sense but the optimizer will do it. I'm not sure if we run the optimizer before or after this pass though.